helm/sealed-secrets/templates/networkpolicy.yaml

```
@@ -10,5 +10,22 @@ spec:
     matchLabels: {{- include "sealed-secrets.matchLabels" . | nindent 6 }}
   ingress:
     - ports:
-      - port: 8080
+      - port: {{ .Values.service.port }}
+  {{- if .Values.networkPolicy.egress.enabled }}
+  egress:
+    - to:
+      {{- if not .Values.networkPolicy.egress.kubeapiCidr }}
+      {{- $kubernetesEndpoint := lookup "v1" "Endpoints" "default" "kubernetes" -}}
+      {{- range $kubernetesAddress := (first $kubernetesEndpoint.subsets).addresses}}
```

first of all, thanks again for your contributions @kvandenhoute . I am testing your MR and I am reproducing something really strange, enabling the egress and generating the template:

```
Error: template: sealed-secrets/templates/networkpolicy.yaml:19:39: executing "sealed-secrets/templates/networkpolicy.yaml" at <first $kubernetesEndpoint.subsets>: error calling first: runtime error: invalid memory address or nil pointer dereference
```
Lookup is going to return a map when you are installing the Sealed Secrets, if you generate just the template the map will be empty. Then the range will try to loop on  nil values. The easy way is to verify if the `$kubernetesEndpoint` is empty or not doing an if before to do the range or something like this:

```suggestion
      {{- if $kubernetesEndpoint }
      {{- range $kubernetesAddress := (first $kubernetesEndpoint.subsets).addresses}}
```

---

That was indeed an oversight. I have fixed the issue, thx for the review :)